master_discovery_fkie/src/master_discovery_fkie/master_discovery.py

```diff
@@ -1089,3 +1119,26 @@ def rosservice_refresh(self, req):
                 import traceback
                 traceback.print_exc()
         return []
+
+
+    def log_message_counters(self):
+        with self.__lock:
+            try:
+                current = self.msg_counter_sent_group
+                delta = current - self.msg_counter_sent_group_last
+                if delta > 0:
+                    rospy.loginfo("UDP Messages sent to group: total=%s, delta=%s", str(current), str(delta))
+                self.msg_counter_sent_group_last = current
+                for k in self.msg_counter_sent_address:
+                    current = self.msg_counter_sent_address[k]
+                    last = self.msg_counter_sent_address_last.get(k,0)
+                    delta = current - last 
+                    if delta > 0:
+                        rospy.loginfo("UDP Messages sent to '%s': total=%s, delta=%s", 
+                                      str(k), str(current), str(delta))
+                    self.msg_counter_sent_address_last[k] = current
```

Comment:
Where are the records removed from `self.msg_counter_sent_address` and `self.msg_counter_send_address_last`?
If not removing them, you can have stale data that you are reporting on.